anvil/src/eth/backend/mem/inspector.rs

```diff
@@ -7,15 +7,19 @@ use crate::{
 use bytes::Bytes;
 use ethers::types::{Address, Log, H256};
 use foundry_evm::{
+    call_inspectors,
     decode::decode_console_logs,
     executor::inspector::{LogCollector, Tracer},
     revm,
-    revm::{CallInputs, EVMData, Gas, Return},
+    revm::{CallInputs, EVMData, Gas, GasInspector, Return},
 };
+use parking_lot::RwLock;
+use std::sync::Arc;
 
 /// The [`revm::Inspector`] used when transacting in the evm
 #[derive(Debug, Clone, Default)]
 pub struct Inspector {
+    pub gas: Option<Arc<RwLock<GasInspector>>>,
```

Comment:
I see this was added in the revm PR

why do we need to guard this via `RwLock` here?

all `Inspector` functions are `mut`

ah is this because `GasInspector` is itself a Inspector, that is also used by the `Tracer` inspector?

since all of them are executed in sequence, there can be no race condition, so perhaps this could be a `Rc<Refcell<>>`?

or we move the `GasInspector` inside the `Tracer`, but then we don't have a standalone `GasInspector`

there's probably not much overhead in `gas_inspector.write` but could add up since this is called per step

Reply:
> but then we don't have a standalone GasInspector

yep, I wanted to call it in a more explicitly way without behaviour hidden in `Tracer` + that way it can be shared between Inspector (I suspect we might need it in the `Debugger` later)

> since all of them are executed in sequence, there can be no race condition, so perhaps this could be a Rc<Refcell<>>?

agree, fixed